test/common.js

```diff
@@ -15,7 +15,8 @@ export const getData = (data, msDelay = GET_DATA_DELAY) => (
 )
 
 export function getDataWithCount({data}, msDelay) {
-  return getData({data: data + ++getDataWithCount.counter}, msDelay)
+  let delay = msDelay instanceof Array ? msDelay[getDataWithCount.counter] : msDelay
```

Comment:
I wouldn't pollute this function with responsibilities just because a single test would find it useful.

This function was nice and simple as it should - gets data with some delay. I'd find it nicer if the test itself created a wrapper function for this:
```
timeoutingGetData = (data) => getDataWithCount(data, [...delays...][getDataWithCount.counter])
```

Reply:
Agreed, fixed